ethereum/weaksubjectivity/src/main/java/tech/pegasys/teku/weaksubjectivity/WeakSubjectivityValidator.java

```diff
@@ -122,35 +125,41 @@ public void validateLatestFinalizedCheckpoint(
       LOG.debug(
           "Latest finalized checkpoint at epoch {} is prior to weak subjectivity checkpoint at epoch {}. Defer validation.",
           latestFinalizedCheckpoint.getEpoch(),
-          maybeWsCheckpoint.orElseThrow().getEpoch());
+          config.getWeakSubjectivityCheckpoint().orElseThrow().getEpoch());
       return;
     }
 
-    // Determine validity
-    boolean isValid = true;
-    if (isAtWSCheckpoint(latestFinalizedCheckpoint)) {
-      // Roots must match
-      isValid = isWSCheckpointRoot(latestFinalizedCheckpoint.getRoot());
+    // Validate against ws checkpoint
+    if (isAtWSCheckpoint(latestFinalizedCheckpoint)
+        && !isWSCheckpointRoot(latestFinalizedCheckpoint.getRoot())) {
+      // Finalized root is inconsistent with ws checkpoint
+      handleInconsistentWsCheckpoint(latestFinalizedCheckpoint.getBlock());
     }
-    isValid = isValid && isWithinWSPeriod(latestFinalizedCheckpoint, currentSlot);
-
-    // Handle invalid checkpoint
-    if (!isValid) {
-      final int activeValidators =
-          calculator.getActiveValidators(latestFinalizedCheckpoint.getState());
-      for (WeakSubjectivityViolationPolicy policy : violationPolicies) {
-        policy.onFinalizedCheckpointOutsideOfWeakSubjectivityPeriod(
-            latestFinalizedCheckpoint, activeValidators, currentSlot);
-      }
+
+    // Determine whether we should suppress ws period errors
+    UInt64 currentEpoch = compute_epoch_at_slot(currentSlot);
+    final Optional<UInt64> suppressionEpoch = getSuppressWSPeriodChecksUntilEpoch(currentSlot);
+    final boolean shouldSuppressErrors =
+        suppressionEpoch.map(e -> e.isGreaterThan(currentEpoch)).orElse(false);
+
+    // Validate against ws period
+    final boolean withinWSPeriod = isWithinWSPeriod(latestFinalizedCheckpoint, currentSlot);
+    if (!withinWSPeriod && !shouldSuppressErrors) {
+      handleFinalizedCheckpointOutsideWSPeriod(latestFinalizedCheckpoint, currentSlot);
+    } else if (!withinWSPeriod
+        && currentSlot.mod(SUPPRESSION_WARNING_PERIOD_IN_SLOTS).equals(UInt64.ZERO)
+        && getAndSetLastLoggedSlot(currentSlot).isLessThan(currentSlot)) {
+      LOG.warn(
+          "Suppressing weak subjectivity errors until epoch {}", suppressionEpoch.orElseThrow());
```

Comment:
I'm not quite sure if this warn would be visible in console? If not may be it makes sense to print to console as well?

Reply:
hmm - good point